### ISSUE TITLE
Stackdriver plugin - data error after a while - every 2 hours 

### ISSUE THREAD

**Issue by Abrams88** [grafana/grafana#16752]
Hi, 

I am using Stackdriver plugin to monitor my compute instances on my GCP project. 
I have connected my Stackdriver data source by using a JWT key like the official documentation says. I have followed all the process and everything works fine. 
However, every 2hours all the panels become red with an ! explanation mark, saying data error. 

If I do a refresh then everything comes back to normal. I have increase and expand all the session and cookies timeouts. 

Can you please explain me why this is happening and if anyone has faced similar behaviour.
I cannot find the source of the issue. 

Any suggestions are welcome please. 

Thanks

**Comment by marefr** [comment]
Grafana version? Any interesting information in Grafana server log when this error happen? Please include full request/response of a query when this error happens - you can use panel query inspector or chrome developer tools/network tab or similar to get the full response of a metric request.

**Comment by Abrams88** [comment]
Dear @marefr , thanks for your reply.
Let me explain a bit more about my situation. 
I am using Grafana version 5.4.3. 
I have installed the Stackdriver plugin and by following the official documentation its connected to my GCP project by using a JWT key. 
The connection works fine.
All the graphs shows values and I can see from the network tab as you suggested the queries every 30sec (as I have configured for refresh) return status code 200. 

The issue:
I am login as admin user and every 1 ~1 and 10mins the graphs goes to red shows an error data. 
The grafana logs that time shows:

> msg="Database table locked, sleeping then retrying" 
> msg="Got new access token" logger=data-proxy-log ExpiresOn=2019-04-25T14:03:52+0000

So, when that is happening, the last query " to _/api/tsdb/query_ send status 200" and the next one send 401
Then I  can see this:   _o/oauth2/v2/auth?login_hint_ status code 405

To access the grafana I pass through Cloudflare. 
I cant find where the issue starts and how this problem is triggered. 

Take a copy of my **config** file:
```
[security]
login_remember_days = 365

[server]
http_port = 3002

[session]
provider = memory
session_life_time = 30585600
cookie_name = grafana_sess
cookie_secure = false

[users]
login_maximum_inactive_lifetime_days = 360
login_maximum_lifetime_days = 360
token_rotation_interval_minutes = 21600

[database]
max_idle_conn = 3
max_open_conn = 3
conn_max_lifetime = 30585600

[auth]
token_rotation_interval_minutes = 1200
```
By refreshing the page everything goes back to normal.
However, its asking to login again...I dont understand why, since the config file keeps the session and cookies. 

Can you please also advice me on that, I want to never expire the session or the cookies that why I have set so high values on days and seconds. Why says that about the access token, I thought that my config file had configured that to not expire in 1 hour. 

Many thanks!

**Comment by marefr** [comment]
Given the [users] and [auth] sections in configuration it looks like you're on Grafana v6.0.

**Comment by Abrams88** [comment]
Dear @marefr ,

You got a very good point on that, I just realised that. I found those configs around in threads in a trial to 
extend the session timeout and cookies. So that means that I should remove those since does not have affect? 
However can those create an issue? 
The config fields in the [session] and [security] tags, affect the session expiration time or not? 
However, my version is for sure 5.4.3. I am using that version because I have installed grafana through puppet modules. 

Thanks for pointing that.
